Feature/add coralogix scaler

YonatanKiron opened the pull request:
<!-- Thank you for contributing!

     Read more about how you can contribute in our contribution guide:
     https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md
-->

_Provide a description of what has been changed_
As many users use Coralogix as their metrics backend, they would like to autoscale their cluster based on the metrics they have in Coralogix.
As Coralogix gateway wraps Prometheus, in the PR we are wrapping the Prometheus Scaler

Things that we would like to add in the next iteration is *Log Query* based scaler

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [ ] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

<!--
  Make sure to link the related issue for this change
  If it requires multiple PRs and/or a PR on another repo as well, please use "Relates to" instead.
-->
Fixes #

<!--
  Make sure to link the related PRs for changes such as documentation & Helm charts
-->
Relates to #

YonatanKiron committed:
Add Coralogix scaler

Signed-off-by: Yonatan Kiron <yonatan.kiron@coralogix.com>

ShauliSolomovich commented on meta.yaml:
```diff
@@ -0,0 +1,6 @@
+name: keda
+deployable_type: image
+lifecycle_status: development
```
you can add `version: 0.1.0` if you want, so it will be easier to use instead of commit id

dependabot[bot] committed:
chore(deps): bump aquasecurity/trivy-action from 0.12.0 to 0.16.0 (#5332)

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.12.0 to 0.16.0.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/0.12.0...0.16.0)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action from 2 to 3 (#5334)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2 to 3.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump dkershner6/reaction-action from 1 to 2 (#5331)

Bumps [dkershner6/reaction-action](https://github.com/dkershner6/reaction-action) from 1 to 2.
- [Release notes](https://github.com/dkershner6/reaction-action/releases)
- [Commits](https://github.com/dkershner6/reaction-action/compare/v1...v2)

---
updated-dependencies:
- dependency-name: dkershner6/reaction-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump tj-actions/branch-names from 7 to 8 (#5333)

Bumps [tj-actions/branch-names](https://github.com/tj-actions/branch-names) from 7 to 8.
- [Release notes](https://github.com/tj-actions/branch-names/releases)
- [Changelog](https://github.com/tj-actions/branch-names/blob/main/HISTORY.md)
- [Commits](https://github.com/tj-actions/branch-names/compare/v7...v8)

---
updated-dependencies:
- dependency-name: tj-actions/branch-names
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

similark committed:
fix isRemovingFinalizer in webhook (#5317)

Signed-off-by: Or Koren <or.koren@similarweb.com>
